### PR TITLE
docs: Update agent autoauth sinks examples

### PR DESCRIPTION
**Pull request by mpalmi**
This PR aims to clarify some confusion around the differing `sinks` specifications in the auto-auth config. Relevant discussion in #7380.

### CODE REVIEW

File: website/content/docs/agent/autoauth/index.mdx [hunk 1]
```diff
@@ -221,5 +259,87 @@ auto_auth {
       path = "/tmp/file-foo"
     }
   }
+
+  sink {
+    type = "file"
```
Comment (hghaf099) [inline website/content/docs/agent/autoauth/index.mdx:264]: Would it worth explaining that you can have multiple sinks of the same type in a single stanza? I assume it is since you added it, but maybe it would a good note to add.

File: website/content/docs/agent/autoauth/index.mdx [hunk 2]
```diff
@@ -198,11 +198,49 @@ These configuration values are common to all Sinks:
...
   sidebar for information about each sink.
 
-### Auto Auth Example
+### Auto Auth Examples
+
+Auto-Auth configuration objects take two separate forms when specified in `HCL`
+and `JSON`. The following examples are meant to clarify the differences between
+the two formats.
+
+#### HCL sinks
```
Comment (hghaf099) [inline website/content/docs/agent/autoauth/index.mdx:207]: Would it be better to rephrase this to `sinks in HCL format`?

File: website/content/docs/agent/autoauth/index.mdx [hunk 3]
```diff
@@ -221,5 +259,87 @@ auto_auth {
...
+      path = "/tmp/file-bar"
+    }
+  }
 }
 ```
+
+~> The [corresponding JSON format](#json-sinks) _must_ specify a `"sinks" :
+[...]` array to encapsulate all `sink` JSON objects.
+
+#### JSON sinks
```
Comment (hghaf099) [inline website/content/docs/agent/autoauth/index.mdx:276]: Would it be better to rephrase this to `sinks in JSON format`? JSON sink may be a confusing title.

**Review by hghaf099 — APPROVED** [review]
lgtm, just added some small suggestions.

**Review by taoism4504 — APPROVED** [review]

File: website/content/docs/agent/autoauth/index.mdx (hunk 2 above)
Comment (mpalmi) [reply website/content/docs/agent/autoauth/index.mdx]: Good call, thanks!
